nvflare/apis/impl/wf_comm_server.py

```diff
@@ -196,7 +197,10 @@ def _do_process_task_request(self, client: Client, fl_ctx: FLContext) -> Tuple[s
         client_task_to_send = None
         with self._task_lock:
             self.logger.debug("self._tasks: {}".format(self._tasks))
-            for task in self._tasks:
+            for t in self._tasks:
+                task = copy.copy(t)
+                task.data = copy.deepcopy(t.data)
```

Comment:
Please do NOT do this. If the task data is huge, then deep copying it would be very inefficient.
Server filters are not used much, we shouldn't sacrifice everything for that small issue.
If a filter needs to change the data, the the filter needs to make deep copy there.

Reply:
Moved the copy so it's only done when filters are applied.